### ISSUE TITLE
XML failure on some F70 EXR images

### ISSUE THREAD

**Issue by Doug-Pardee** [Doug-Pardee/LightZombie#7]
slapp0 is getting XML failures on some F70 EXR images and not others. Failure occurs on trying to save the file.

Sample files can be found in the download area:
Fails: [DSCF3187 problematic.JPG](https://github.com/downloads/Doug-Pardee/LightZombie/DSCF3187%20problematic.JPG)
Works: [DSCF3191 working.JPG](https://github.com/downloads/Doug-Pardee/LightZombie/DSCF3191%20working.JPG)

**Comment by krisw** [comment]
I have tried to download both files, DSCF3187 appears to download but does not show in Downloads folder but DSCF3191 does download.

I will test both files for slappo if the download can be sorted

**Comment by Doug-Pardee** [comment]
I've tested both files on Windows, and I was able to open the file, add a Relight tool, save the LZN-JPEG, and open the LZN-JPEG on both.

I reran that test after replacing the Java 6u2 JRE shipped with LZ 3.9 with the 6u29 JRE, and it still ran fine. The LightZone.log file confirms it's running with 6u29.

So that doesn't help.

I took a look at the XML metadata that was inserted by LightZone, and it looked fine. There was an APP4 segment with the correct 2-byte length, then the XML:

```
<LightZoneTransform xmlns="http://www.lightcrafts.com/lightzone/LightZoneTransform" version="8">
  <Scale Factor="0.39699075"/>
  <Image orientation="1" path="E:\LightZone3\Repackage\Test\XML-failure\DSCF3187 problematic.JPG" relativePath="DSCF3187 problematic.JPG"/>
  <Controls>
    <GenericOperation Active="true" Collapsed="false" Locked="false" Mode="Normal" Name="Relight 1" Opacity="100" OperationType="Tone Mapper V5" layerControlsIndex="0" regionsInverted="false">
      <ColorSelection HueBlue="0.5" HueEnabled="true" HueGreen="0.5" HueRadius="-1.0" HueRed="0.5" Inverted="false" LuminosityEnabled="true" LuminosityLower="0.0" LuminosityLowerFeather="0.0" LuminosityUpper="1.0" LuminosityUpperFeather="0.0"/>
      <Slider Depth="64.0" Detail="1.5" Fuzz="0.1" Highlights="0.2" Shadows="3.0"/>
      <Checkbox/>
      <Choice/>
    </GenericOperation>
    <Region/>
    <Crop Angle="0.0"/>
  </Controls>
  <Save type="JPEG (sidecar)">
    <ExportOptions file="E:\LightZone3\Repackage\Test\XML-failure\DSCF3187 problematic_lzn.jpg" type="JPEG">
      <com.lightcrafts.image.export.OriginalWidthOption value="0"/>
      <com.lightcrafts.image.export.OriginalHeightOption value="0"/>
      <com.lightcrafts.image.export.BlackPointCompensationOption value="false"/>
      <com.lightcrafts.image.export.ColorProfileOption value="sRGB IEC61966-2.1"/>
      <com.lightcrafts.image.export.RenderingIntentOption value="0"/>
      <com.lightcrafts.image.export.ResizeWidthOption value="1024"/>
      <com.lightcrafts.image.export.ResizeHeightOption value="1024"/>
      <com.lightcrafts.image.export.ResolutionOption value="300"/>
      <com.lightcrafts.image.export.ResolutionUnitOption value="2"/>
      <com.lightcrafts.image.export.QualityOption value="85"/>
    </ExportOptions>
  </Save>
</LightZoneTransform>
```

This being Windows, each line (including the last) was termined by a CR/LF pair.

**Comment by Doug-Pardee** [comment]
I took a look at what was changed in the 3.9.2 release for the Java SE6 update, and it appears to have been just a bug fix in the setting-up of the Quaqua "look and feel" for the user interface. Nothing XML-related that I can see.

**Comment by Doug-Pardee** [comment]
slapp0 sent me a lightzone.log file that should prove helpful. One thing I noticed is that the path to his file includes a "Swedish" character (Š). LightZone writes the path to the "master" image file into the XML in the LZN-JPEG, and that character would need to be translated into a 2-byte UTF-8 sequence.

This could be what's triggering the bug. If passed through without proper translation to UTF-8, the Š would come out as a 138 (0x8A), and that definitely would fit the "Invalid byte 1" error message because the first byte of a UTF-8 sequence can't be in the ranges 128-191 or 245-255. I'll check this out on Windows when I get the chance, and I've asked slapp0 to try moving the problematic file to a file path containing only "English" characters.

**Comment by Doug-Pardee** [comment]
Here's a "shotgun" attempt at a fix, based solely on the assumption that Mac Java isn't defaulting to UTF-8 encoding. I have no idea if this will work or not.

Edit the `Info.plist` file (located in the `LightZone.app/Contents` folder), and look for this down near the bottom:

```
    <key>Properties</key>
    <dict>
        <key>apple.awt.graphics.UseQuartz</key>
        <string>true</string>
        <key>com.lightcrafts.licensetype</key>
        <string>ESD</string>
    </dict>
```

Add two lines to it, to make it like this:

```
    <key>Properties</key>
    <dict>
        <key>apple.awt.graphics.UseQuartz</key>
        <string>true</string>
        <key>com.lightcrafts.licensetype</key>
        <string>ESD</string>
        <key>file.encoding</key>
        <string>utf-8</string>
    </dict>
```

**Comment by Doug-Pardee** [comment]
slapp0 reports that both of the above worked. Using pure-ASCII paths and filename allowed previously unsavable files to be saved, and the change to `Info.plist` allowed previously unsavable files to be saved in paths containing non-ASCII characters.

This would appear to be LightZone erroneously using the "default charset", rather than specifying UTF-8, somewhere in the generation of the XML text to be saved into the LZN-JPEG file.

I've added an [updated `Info.plist` file](https://github.com/downloads/Doug-Pardee/LightZombie/Info.plist) to the available downloads here on the github project.

**Comment by krisw** [comment]
Doug

Thanks for info on how this issue has developed and that it now appears to be sorted.

It ties in with what I always suspected arising from a similar occurrence I had some years ago. I got over it somehow for a particular file but have forgotten how I did it. From that time I took evasive action by having my image files for Lightzone in a dedicated library untouched by sources out of my control or other editing programs. Since my RAW images are sourced directly from my Canon 20D and now my 7D, or as TIF files from my slides or old family photographs scanned in my Minolta Scan Elite 5400 or Epson 4490 scanners, I know they are not touched by extraneous metadata which might cause problems.

I have added your change into my Info.plist.

Prior to this I downloaded slapp0's "good" file but was unable to download his "bad" file to test on my system. The "good" could be edited without issue. There is a question as to why the "bad" file did not download properly from GitHub as an image but as a text file which I deleted and cannot recover. I now find the file inaccessible on GitHub - I assume you have taken both those files off.

It seems to me that LightZombie / GitHub are doing well by LZ and its users. Well done!

Kris
